### PR TITLE
Use lume, overhaul frontpage and add a target list

### PR DESCRIPTION
**Pull request by Yatekii**

### CODE REVIEW

File: src/targets/index.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+layout: layouts/targets.jsx
+title: targets
+mainMenu: targets
+description: |
+  **probe-rs** ships many in-built targets.
```
Comment (Tiwalun) [inline src/targets/index.yaml:5]:
```suggestion
  **probe-rs** ships many built-in targets.
```

File: src/targets/index.yaml
Comment (Tiwalun) [inline src/targets/index.yaml:1]: - Links to targets are broken, return a 404 page
- Impossible to figure out what the dropdowns mean:
  image

File: src/targets/pages.tmpl.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+export const layout = "layouts/target.jsx";
+
+export default async function* ({ loadTargets, latestRelease }) {
+  // TODO:
+  return;
```
Comment (Tiwalun) [inline src/targets/pages.tmpl.js:5]: Something missing here?

File: src/docs/getting-started/probe-setup.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+description: "How to set up your debug probe to work with probe-rs."
+order: 20
+---
+
+The probe-rs project supports three probes out of the box.
+
+Most of them require little to no setup. For linux, [udev](#udev-rules) rules
+are required for non admin access.
+
+The [ST-Link](#st-link), to [J-Link](#segger-j-link) and [CMSIS-DAP](#cmsis-dap)
```
Comment (Tiwalun) [inline src/docs/getting-started/probe-setup.md:12]:
```suggestion
The [ST-Link](#st-link), [J-Link](#segger-j-link) and [CMSIS-DAP](#cmsis-dap)
```

File: src/docs/getting-started/probe-setup.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+based probes are supported.
+
+## udev rules
+
+By default, the debug probes are only accessible by users with root privileges
+on Linux based systems. It is recommend to use appropriate udev rules to allow
+users without root privileges access to the debug probes as well.
+
+1. Download the file [69-probe-rs.rules](/files/69-probe-rs.rules)[^1] and place
```
Comment (Tiwalun) [inline src/docs/getting-started/probe-setup.md:21]: The [^1] doesn't seem to work, and the link should download the file, not open it in  a new window.

**Review by Tiwalun — COMMENTED** [review]
- Links to targets are broken, return a 404 page.

File: src/targets/pages.tmpl.js (hunk 2 above)
Comment (Yatekii) [reply src/targets/pages.tmpl.js:5]: yep, this is why the links to targets are broken. I dont generate any target pages during development for faster page generation :) Will fix :)

File: src/docs/getting-started/probe-setup.md (hunk 4 above)
Comment (Tiwalun) [reply src/docs/getting-started/probe-setup.md:21]: Use something like `<a href=xxx download>69-probe-rs.rules</a>`? Or do fancy javascript frameworks not support this :stuck_out_tongue: ?

Comment (Yatekii) [reply src/docs/getting-started/probe-setup.md:21]: Oh there is a download hint? Did not know!

**Review by adamgreig — APPROVED** [review]

**Pull request merged**